### ISSUE TITLE
Deployment validation pipeline with an automatic trigger from a test/dev branch

### ISSUE THREAD

**Issue by mkaalto** [awslabs/landing-zone-accelerator-on-aws#400]
**Is your feature request related to a problem?**
At present, we don't have a validation mechanism for AWS Landing Zone Accelerator deployments before they are pushed onto the main production branch. This poses issues as any potential problems are often diagnosed late, and it also poses the risk of disrupting our production environment. It is also slow to run the whole pipeline in the development stages. Local validator installation additionally requires extra work for each person doing the development.

**Describe the feature you'd like**
I propose the creation of a dedicated AWS Landing Zone Accelerator deployment/validation pipeline. The pipeline would handle all predefined validations without interfering with our production system. The key element of this feature would be the automated triggering system; it should activate whenever a push is made to a non-main (dev/test) branch. Only after the validation process would the completed features could be merged into the main production branch.

The new pipeline would serve the purpose of:

* Automating deployment/validation process on test/dev branches
* Pre-checking for any disruptable changes before affecting the main branch
* Improving our deployment efficiency and overall reliability

**Additional context**
As we strive to automate as much process in our DevOps life cycle, integrating this feature will significantly increase our productivity and minimize errors.

Apart from the AWS Landing Zone Accelerator, I’m also open to explore other alternative solutions like AWS Control Tower, AWS CDK / CloudFormation or even third-party tools that can address the mentioned problem efficiently. This could involve altering the current process or implementing a new setup.

Feedback from the team would be greatly appreciated. What are your thoughts about the proposed solution? Are there alternative options or best practices we might consider as well?

**Comment by Athanasii** [comment]
Hello @mkaalto , thank you for filing a feature request with the Landing Zone Accelerator team! Our team will evaluate the applicability of this request for a future release. 

We will keep this issue open for tracking purposes and keep you aware of any status updates going forward. Thank you for supporting Landing Zone Accelerator!
